### PR TITLE
Change operations to fix DBAL error

### PR DESCRIPTION
**Pull request by zonky2**
Change operations to fix DBAL error
see #549

### CODE REVIEW

**Review by discordier — APPROVED** [review]

File: src/Data/DefaultDataProviderDBalUtils.php [hunk 1]
```diff
@@ -100,6 +101,10 @@ public static function addSorting(ConfigInterface $config, QueryBuilder $queryBu
         }
 
         foreach ($config->getSorting() as $sort => $order) {
+            if (empty($sort)) {
```
Comment (baumannsven) [inline src/Data/DefaultDataProviderDBalUtils.php:104]: Why test for empty `$sort`. Can you explain that.

Comment (zonky2) [reply src/Data/DefaultDataProviderDBalUtils.php:104]: frag den Meister @discordier ;-)

Comment (baumannsven) [reply src/Data/DefaultDataProviderDBalUtils.php:104]: @discordier can you explain that? The key of an foreach loop can not be should empty.

Comment (baumannsven) [reply src/Data/DefaultDataProviderDBalUtils.php:104]: If so, should a warning be triggered?

Comment (discordier) [reply src/Data/DefaultDataProviderDBalUtils.php:104]: In addition to the exception or instead?
If we opt for exception, no warning, if we opt for ignoring, we should trigger a warning.

Comment (baumannsven) [reply src/Data/DefaultDataProviderDBalUtils.php:104]: I would be in favour of ignoring that and trigger the warning.

**Review by baumannsven — APPROVED** [review]

**Comment by baumannsven** [comment]
Merged in ab03ae05a97cbe9c1618da24457e9b26c09a4c4a

**Pull request merged**